### ISSUE TITLE
AI to improve developer experience on xrpl.org

### ISSUE THREAD

**Issue by generalburatai** [XRPLF/xrpl-dev-portal#2090]
**AI to improve developer experience on xrpl.org:**
In H2 2023, we propose to enable a learning journey for XRPL developers via Artificial Intelligence (AI) functionality on xrpl.org with a chat-to-search bot. This work will (1) shorten developer learning journey, (2) help developers to get educated on the xrpl capabilities, and (3) decipher steps to build on the ledger, enhancing the developer experience as they onboard through xrpl.org. At launch, a developer will be able to get on any page on xrpl.org and ask a question to AI in a chat-to-search bot in plain English text and the developer will receive synthesized information that the AI gathered from model trained on xrpl.org Documentation and related pages. Please find attached the design specs for xrpl.org.

Our goal is to deliver a first version of a ‘chat to search’ bot  to aid in information synthesis and drive usability of the xrpl.org by building Functional UI within the xrpl.org in two phases: The first phase is a chatbot that enables developers to ask questions and receive text-like responses. If successful (success defined as developers receiving accurate responses), then we will have the search box on xrpl.org replaced by the chat-to-search UI. Production grade chat-based LLM model will be used to deliver synthesized information on how to use the XRPL. Focus of MVP will be XRPL.org document set, and limited external documentation.

For example, a developer can ask questions to the chat-to-search bot in plain English about how to use the XRPL - “Could you tell me how to use the fundWallet function on XRPL?” and receive responses in plain English and get targeted action oriented feedback on how to do it.
Example response:
“Hello, yes, I’d be happy to provide some information about the fundWallet function that you are referring to. 
fundWallet is a function for you to fund your testnet or devnet wallet from a faucet. To do so, it’s best to go to the Get Started with Javascript page here (https://xrpl.org/get-started-using-javascript.html#3-get-account) 
If you are using Python, here’s the link to get started with https://xrpl.org/get-started-using-python.html#2-get-account
The fundWallet function in xrpl-py is called generate_faucet_wallet

Try them out and let me know if you have additional questions!”
<img width="1512" alt="Screenshot 2023-08-28 at 12 16 57 AM" src="https://github.com/XRPLF/xrpl-dev-portal/assets/116908031/5d872322-db86-4446-8d03-688a8adbb2ab">

**Comment by tequdev** [comment]
Is support for multiple languages possible?

**Comment by generalburatai** [comment]
> multiple

For this v.1., only English language is supported. Future versions may enable multiple languages.

**Comment by generalburatai** [comment]
> Is support for multiple languages possible?

For this v.1., only English language is supported. Future versions may enable multiple languages.

**Comment by amarantha-k** [comment]
The AI Chatbot in Beta preview was added by https://github.com/XRPLF/xrpl-dev-portal/pull/2207.
